pyrobosim/pyrobosim/core/robot.py

```diff
@@ -123,6 +156,24 @@ def set_path_executor(self, path_executor):
             return
         path_executor.robot = self
 
+    def is_moving(self):
+        """
+        Checks whether the robot is moving, either due to a navigation action or velocity commands.
+
+        :return: True if the robot is moving, False otherwise.
+        :rtype: bool
+        """
+        return self.executing_nav or np.count_nonzero(self.dynamics.velocity) > 0
```

Comment:
A `get_current_velocity()` method in the `RobotDynamics2D` could be useful here, instead of directly accessing the velocity attribute.

Reply:
I think the `RobotDynamics2D` instance can be considered a part of the `Robot` class... so I'm OK leaving this to just directly access velocity here, as is done in `Robot.get_pose()` which directly accesses `robot.dynamics.pose`.